components/checkbox/src/stories.tsx

```diff
@@ -18,6 +18,12 @@ Default.args = {
   children: 'Example',
 };
 
+export const CheckboxSmall = Template.bind({});
+CheckboxSmall.args = {
+  children: 'Small checkbox',
+  sizeVariant: 'SMALL',
```

Comment:
I think the prop value that is passed in should be lowercase, i.e. it should be usable as `<Checkbox sizeVariant="small" />`

Reply:
@penx I don't mind but this clashes with other parts of your code. All sizes so far have been uppercase.
Wouldn't it be great if we can retro fit the type for `sizeVariant` on all existing component. It will keep everything more manageable from a code point of view?

If you still need it changed to lowercase, I can still do so?

Existing doc:
![Screenshot 2023-03-08 at 20 46 47](https://user-images.githubusercontent.com/1698584/223846161-8e061cc3-ebd9-4f2e-9f78-0598b3946ab1.png)

Reply:
Ah ok. Thanks. Let me have a think

Reply:
> Ah ok. Thanks. Let me have a think

@penx do you have an update. 

I think we should keep the size names consistent with the rest of your code. If we can start making it TypeSafe for now then we can change it much easier later on. I am happy to go with whatever you decide

Reply:
sorry just to let you know I haven't forgotten about this but have had a busy few weeks - will hopefully find some time tomorrow to review

Reply:
ok great, confirmed Heading and Caption currently have a size prop that accepts a number or an uppercase string in this set.

We should be consistent here, and if we make a change it should be done across the library.

We're breaking from convention here and using `sizeVariant` instead of `size` so as not to conflict with an existing html attribute.

So... all good I think!

Are you hoping to see this in a release soon?

Reply:
> Are you hoping to see this in a release soon?

there was a urgency but no so much anymore. so approve and merge whenever you can :)